### ISSUE TITLE
add option -- "Delete all students projects and remove all students from course”?

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#2429]
> thanks, although this isn’t ideal because we have to copy all the material to the new course. For now I’ve deleted all the students manually. Since there is already an option to “Delete all students projects”, perhaps it would be useful to add “Delete all students projects and remove all students from course”?

Requested by: Jean-Marc Schwartz

**Comment by schwartzjmc** [comment]
Hi, has there been any progress on this issue? I encounter it every year at the new term start...

**Comment by johnjeng** [comment]
@schwartzjmc I'm going to take a stab at it this week. Is the real life problem starting an identical course each year with new students but the same assignments/handouts? Or do you actually want to "delete" the students as the functionality exists now (eg. you want to be able to undelete them with all their progress and stuff saved).

**Comment by schwartzjmc** [comment]
@johnjeng Yes the idea is to keep the same course but change the list of students. I can "delete" them one by one, but there doesn't seem to be a way to remove them all from the course in one go?

**Comment by johnjeng** [comment]
@schwartzjmc Thanks! It sounds like you're looking for #1826 but since you've been using delete for this, I'll add the functionality to delete the students all at once (using the current mechanism). This is not actually ideal for solving #1826 but is definitely easier to implement.

Maybe you've already gone and manually deleted the students already but I should have this out tomorrow.

**Comment by schwartzjmc** [comment]
@johnjeng Yes I've done it manually for this year but it will definitely useful to have this function next year, thanks!
